### PR TITLE
Fixed the timers for cutscene-replacement

### PR DESCRIPTION
**Pull request by UltiNaruto**

### CODE REVIEW

**Comment by toasterparty** [comment]
@UltiNaruto this fixes life grove hudmemo? If you put your motivation in your pull request descriptions, it helps me understand and get through the review faster.

**Comment by UltiNaruto** [comment]
It fixes life grove hud memo and should fix Thardus on major cutscene removal

File: src/patches.rs [hunk 1]
```diff
@@ -6448,13 +6452,16 @@ fn patch_remove_cutscenes(
...
                         message:  structs::ConnectionMsg::RESET_AND_START,
-                        target_object_id: timer_id,
+                        target_object_id: timer_ids[0],
                     });
 
                     relay_connections.retain(|conn| conn.state != structs::ConnectionState::INACTIVE);
                     objs_to_add.push(timer);
                 }
 
+                if timer_ids.len() > 0 {
```
Comment (toasterparty) [inline src/patches.rs]:
```rust
let timer_id = timer_ids.pop();
```

is a cleaner way of doing this same thing

**Review by toasterparty — CHANGES_REQUESTED** [review]

File: src/patches.rs (hunk 1 above)
Comment (UltiNaruto) [reply src/patches.rs]: rather pop_back() since I remove the first entry

Comment (UltiNaruto) [reply src/patches.rs]: nvm I'll do it from the last entry then

**Review by toasterparty — APPROVED** [review]

**Pull request merged**